web/src/routes/backOffice/employees/components/EmployeeForm.svelte

```diff
@@ -0,0 +1,491 @@
+	 */
+	const formFieldsLengths = {
+		username: {
+			min: 0,
+			max: 30,
+		},
+		firstName: {
+			min: 0,
+			max: 30,
+		},
+		lastName: {
+			min: 0,
+			max: 30,
+		},
+		phoneNumber: {
+			min: 0,
+			max: 15,
+		},
+	};
+
+	/**
+	 * Error messages of the form fields.
+	 */
+	let formErrorMessages = {
+		username: "",
+		firstName: "",
+		lastName: "",
+		dateOfBirth: "",
+		phoneNumber: "",
+		location: "",
+		scheduleStart: "",
+		scheduleEnd: "",
+	};
+
+	/**
+	 * Validates form and sets error messages on the form fields if they contain errors.
+	 * @param usernameValidity
+	 * @param firstNameValidity
+	 * @param lastNameValidity
+	 * @param dateOfBirthValidity
+	 * @param phoneNumberValidity
+	 * @param locationValidity
+	 * @param coordinate
+	 */
+	function validateForm(
+		usernameValidity: ValidityState,
+		firstNameValidity: ValidityState,
+		lastNameValidity: ValidityState,
+		dateOfBirthValidity: ValidityState,
+		phoneNumberValidity: ValidityState,
+		locationValidity: ValidityState,
+		scheduleStart: ValidityState,
+		scheduleEnd: ValidityState,
+		coordinate: number[] | undefined,
+	): coordinate is number[] {
+		// Username Validation.
+		if (usernameValidity.valueMissing) {
+			formErrorMessages.username = $t("error.valueMissing");
+		} else if (usernameValidity.tooShort) {
+			formErrorMessages.username = $t("error.tooShort", {
+				minLength: formFieldsLengths.username.min,
+			});
+		} else if (usernameValidity.tooLong) {
+			formErrorMessages.username = $t("error.tooLong", {
+				maxLength: formFieldsLengths.username.max,
+			});
+		} else {
+			formErrorMessages.username = "";
+		}
+
+		//FirstName Validation.
+		if (firstNameValidity.valueMissing) {
+			formErrorMessages.firstName = $t("error.valueMissing");
+		} else if (firstNameValidity.tooShort) {
+			formErrorMessages.firstName = $t("error.tooShort", {
+				minLength: formFieldsLengths.firstName.min,
+			});
+		} else if (firstNameValidity.tooLong) {
+			formErrorMessages.firstName = $t("error.tooLong", {
+				maxLength: formFieldsLengths.firstName.max,
+			});
+		} else {
+			formErrorMessages.firstName = "";
+		}
+
+		// LastName Validation.
+		if (lastNameValidity.valueMissing) {
+			formErrorMessages.lastName = $t("error.valueMissing");
+		} else if (lastNameValidity.patternMismatch) {
+			formErrorMessages.lastName = $t("error.patternMismatch");
+		} else if (lastNameValidity.tooShort) {
+			formErrorMessages.lastName = $t("error.tooShort", {
+				minLength: formFieldsLengths.lastName.min,
+			});
+		} else if (lastNameValidity.tooLong) {
+			formErrorMessages.lastName = $t("error.tooLong", {
+				maxLength: formFieldsLengths.lastName.max,
+			});
+		} else {
+			formErrorMessages.lastName = "";
+		}
+
+		// DateOfBirth Validation.
+		if (dateOfBirthValidity.valueMissing) {
+			formErrorMessages.dateOfBirth = $t("error.valueMissing");
+		} else if (dateOfBirthValidity.patternMismatch) {
+			formErrorMessages.dateOfBirth = $t("error.patternMismatch");
```

Comment:
Is there a possibility of a incorrect pattern for dates inputs given that they are not free-form fields?

Reply:
don`t think so